src/pages/index.js

```diff
@@ -0,0 +1,251 @@
+    if (
+      evt.target.classList.contains("modal") ||
+      evt.target.classList.contains("modal__close-button")
+    ) {
+      closeModal(modal);
+    }
+  });
+});
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", handleEscape);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  handleLoadingState(editModalSubmit, true);
+  api.editUserInfo({
+    name:  editModalNameInput.value,
+    about: editModalDescriptionInput.value
+  })
+    .then((data) =>{
+  profileName.textContent = data.name;
+  profileDescription.textContent = data.about;
+  closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() =>{
+      handleLoadingState(editModalSubmit, false)}
+    )};
+
+function handleAddFormSubmit(evt) {
+  evt.preventDefault();
+  handleLoadingState(addCardSubmit, true);
+  api.createCardInfo({
+    link: addModalLinkInput.value ,
+    name: addModalNameInput.value
+  })
+   .then((data) =>{
+    const cardElement = getCardElement(data);
+    cardsList.prepend(cardElement);
+    addFormElement.reset();
+    disableButton(addCardSubmit);
+    closeModal(addCardModal);
+  })
+  .finally(() =>{
+  handleLoadingState(addCardSubmit, false)
+});
+}
+
+function handleAvatarFormSubmit(evt){
+  evt.preventDefault();
+  handleLoadingState(avatarCardSubmit, true);
+  api.editAvatarInfo(avatarModalInput.value)
+  .then((avatar) => {
+    avatarPicture.src = avatar.link;
+    closeModal(avatarModal);
+  })
+  .finally(() =>{
+  handleLoadingState(avatarCardSubmit, false)});
+}
+
+function handleLoadingState(button, isLoading) {
+  button.textContent = isLoading ? "Saving..." : "Save";
+}
+
+function handleDeleteLoadingState(button, isLoading) {
+  button.textContent = isLoading ? "Deleting..." : "Delete";
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(editFormElement, [
+    editModalNameInput,
+    editModalDescriptionInput,
+  ], validationConfig);
+  openModal(editModal);
+});
+
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+
+addFormElement.addEventListener("submit", handleAddFormSubmit);
+
+avatarFormElement.addEventListener("submit", handleAvatarFormSubmit);
+
+newPostButton.addEventListener("click", () => {
+  openModal(addCardModal);
+});
+
+avatarEditButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+cardModalCancel.addEventListener("click", () => {
+  closeModal(cardDeleteModal) });
+
+cardModalConfirm.addEventListener("click", () => {
+  handleDeleteLoadingState(cardModalConfirm, true);
+  api.deleteCardInfo(cardToDelete.id)
+  .then(() => {
+   cardToDelete.element.remove();
+   closeModal(cardDeleteModal);
+  })
+  .catch(console.error)
+  .finally(() => {
+   handleDeleteLoadingState(cardModalConfirm, false);
+  });
+   });
+
```

Comment:
* You need to **format** the markup (and code) to show **indentation** and to make the code more readable. You can do that with `Prettier` or a combination: press `SHIFT + ALT + F`